src/table.rs

```diff
@@ -33,6 +40,10 @@ pub enum Item {
     Table(Table),
     /// Type representing array of tables.
     ArrayOfTables(ArrayOfTables),
+    /// Type for display walker to know where (during walk) to
+    /// print dotted key. To get value, parse
+    /// the key and descend table tree.
+    DottedKeyMarker(Vec<SimpleKey>),
```

Comment:
i don't like that the internal implementation detail is leaking into the public API

Reply:
Should we make Item opaque to the user?

Reply:
Hmm, I'm not sure. Currently, it's possible to delete an item using `table['key'] = Item::None` and I'd like to avoid unnecessary breaking changes unless there is a good reason for that. Again, let's think about what kind of API we'd like to expose the user (primarity cargo-edit), and go from that.